### ISSUE TITLE
Images scanning on colab is super slow

### ISSUE THREAD

**Issue by MosbehBarhoumi** [ultralytics/ultralytics#18253]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Other

### Bug

I can't believe we are required to scan all images during training YOLO models, especially on platforms like Google Colab. Why isn't there an option to disable this scanning step? It’s taking an excessive amount of time, and after processing hundreds of images, the process just stops completely.

Please don’t point out I/O bottlenecks when accessing data from Google Drive I’m fully aware of that. My question is: **why isn’t there a way to skip this scanning process altogether?**
![Screenshot from 2024-12-16 08-52-59](https://github.com/user-attachments/assets/935f571a-fa00-4d47-832e-d35ef76fe22e)

### Environment

data on drive and train any yolo version.

### Minimal Reproducible Example

data on drive and train any yolo version.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @MosbehBarhoumi, thank you for your interest in Ultralytics 🚀! We understand your frustration and appreciate you bringing this up.

We recommend checking out the [Docs](https://docs.ultralytics.com) for the latest tutorials and references, such as [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples, as they might provide insights or alternative workflows. These resources often address common questions and challenges faced by our users.

Since this appears to be a 🐛 Bug Report or feedback around performance, we kindly request you to provide a formal [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) (MRE) to help us investigate this further. For example:
- What specific Ultralytics YOLO version you are using (`ultralytics.__version__`)
- Steps to reproduce the issue
- An example of how the image scanning is behaving in your workspace compared to your expectations.

For better performance on Google Colab, please ensure:
1. Your `ultralytics` package is fully upgraded with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment:
   ```bash
   pip install -U ultralytics
   ```
2. Your dataset I/O bottlenecks are minimized. Using a local dataset or hosting data on a faster cloud service may significantly reduce latency compared to Google Drive.

For real-time discussions or additional support, join the Ultralytics community:
- [Discord](https://discord.com/invite/ultralytics) 🎧 (best place for immediate help)
- [Discourse](https://community.ultralytics.com)
- [Subreddit](https://reddit.com/r/Ultralytics)

## Environments

YOLO supports running in the following up-to-date verified environments, offering free or readily available GPUs:
- **Google Colab:** <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a>
- **Paperspace Notebooks:** <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a>
- **Kaggle Notebooks:** <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

This is an automated response to help you get started while we analyze your issue. An Ultralytics engineer will review your report and respond shortly. Thank you for your patience and for helping make Ultralytics better! 😊

**Comment by Y-T-G** [comment]
It wouldn't take that long per image. What's the training command you're using?

**Comment by Y-T-G** [comment]
Can you provide the output after running this command: `!yolo checks`?

**Comment by MosbehBarhoumi** [comment]
> It wouldn't take that long per image. What's the training command you're using?

Well I took the screenshot after it stopped for a while for no reason but still slow!

**Comment by Y-T-G** [comment]
It should be a lot faster since the scan is simply opening and reading text files.

**Comment by RizwanMunawar** [comment]
@MosbehBarhoumi Thank you for bringing up the issue. The problem can be resolved by increasing the `workers` parameter, such as setting `workers=32` or `workers=64`. This adjustment updates the final training command to something like:

```
results = model.train(data=data_yaml_path, epochs=100, imgsz=640, project=base_dir, 
name="yolo_training", workers=32)
```

Please try it and let me know your thoughts. Thank you!

**Comment by MosbehBarhoumi** [comment]
@RizwanMunawar I have tried that,  the only method that works, which involves copying images from Drive to local Colab files. However, this is not a good solution because every time I close the Colab tab, I lose the data stored in the local Colab files.
My main question is: why is there no option to disable scanning?

**Comment by MosbehBarhoumi** [comment]
@Y-T-G The problem is not related to YOLO directly it's about drive-colab, but an option for disabling scanning even it happened once (will be cashed next run) for large datasets can cause a problem.

**Comment by Y-T-G** [comment]
If the read speed is slow for txt files which are just a few KBs, then the training will be much slower because it requires reading images which are usually in MBs, along with the txt files. Disabling scanning will not help much because the slow I/O speed is going to affect your training too.
